From 2f674d980a116075dc7123a3e243b1451e4a732d Mon Sep 17 00:00:00 2001
From: Max Krummenacher <max.oss.09@gmail.com>
Date: Tue, 3 Nov 2015 22:13:40 +0100
Subject: [PATCH] image_util.cc: disable warning from gcc 5

addresses:
  ../../ui/gfx/image/image_util.cc:50:6: error: assuming signed overflow does
  not occur when assuming that (X - c) <= X is always true
  [-Werror=strict-overflow]
  bool VisibleMargins(const ImageSkia& image, int* leading, int* trailing) {
       ^

Signed-off-by: Max Krummenacher <max.oss.09@gmail.com>

Upstream-Status: Pending
See the discussion on the issue in Chromium upstream:
https://groups.google.com/a/chromium.org/forum/#!topic/chromium-dev/3uwBwunIa7g

---
 ui/gfx/image/image_util.cc | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/ui/gfx/image/image_util.cc b/ui/gfx/image/image_util.cc
index 89a3f8c..d595da3 100644
--- a/ui/gfx/image/image_util.cc
+++ b/ui/gfx/image/image_util.cc
@@ -47,7 +47,10 @@
 }
 #endif  // !defined(OS_IOS)
 
+#pragma GCC diagnostic push
+#pragma GCC diagnostic warning "-Wstrict-overflow"
 bool VisibleMargins(const ImageSkia& image, int* leading, int* trailing) {
+#pragma GCC diagnostic pop
   *leading = 0;
   *trailing = std::max(1, image.width()) - 1;
   if (!image.HasRepresentation(1.0))
-- 
1.8.4.5
